feg/gateway/services/eap/providers/aka/servicers/service.go

```diff
@@ -135,32 +136,15 @@ func NewEapAkaService(config *mconfig.EapAkaConfig) (*EapAkaSrv, error) {
 					time.Millisecond * time.Duration(config.Timeout.SessionAuthenticatedMs))
 			}
 		}
-		for _, plmnid := range config.PlmnIds {
-			l := len(plmnid)
-			switch l {
-			case 5:
-				service.plmnIds[plmnid] = plmnIdVal{l5: true}
-			case 6:
-				plmnid5 := plmnid[:5]
-				val, _ := service.plmnIds[plmnid5]
-				val.b6 = plmnid[5]
-				service.plmnIds[plmnid5] = val
-			}
-		}
+		service.plmnIds = plmn_filter.GetPlmnVals(config.PlmnIds)
 	}
 	return service, nil
 }
 
 // CheckPlmnId returns true either if there is no PLMN ID filters (allowlist) configured or
 // one the configured PLMN IDs matches passed IMSI
 func (s *EapAkaSrv) CheckPlmnId(imsi aka.IMSI) bool {
-	if len(s.plmnIds) == 0 {
-		return true
-	}
-	if val, ok := s.plmnIds[string(imsi)[:5]]; ok && (val.l5 || (len(imsi) > 5 && val.b6 == imsi[6])) {
-		return true
-	}
-	return false
```

Comment:
note line 169 shows `imsi[6]`, That was a bug since the 6th character is the one at index 5